decayAmplitude/massDependence.cc

```diff
@@ -564,6 +617,231 @@ piPiSWaveAuMorganPenningtonKachaev::piPiSWaveAuMorganPenningtonKachaev()
+
+		if(not massDepKey->lookupValue("G0", G0)){
+			printErr << "cannot find 'G0' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiF", phiF)){
+			printErr << "cannot find 'phiF' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiR", phiR)){
+			printErr << "cannot find 'phiR' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("phiRsin", phiRsin)){
+			printErr << "cannot find 'phiRsin' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("F", F)){
+			printErr << "cannot find 'F' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("R", R)){
+			printErr << "cannot find 'R' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("MMax", MMax)){
+			printErr << "cannot find 'MMax' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		if(not massDepKey->lookupValue("tag", tag)){
+			printErr << "cannot find 'tag' value in configuration for mass dependence '" << Name() << "'. Aborting..." << endl;
+			throw;
+		}
+
+		return KPiSGLASS::Create(a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax, tag);
+	} else {
+		printWarn << "no configuration given for mass dependence '" << Name() << "'. Using dummy config" << endl;
+		return KPiSGLASS::Create(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, "");
+	}
+}
+
+
+std::string
+KPiSGLASS::parentLabelForWaveName(const isobarDecayVertex& v) const{
+	std::ostringstream label;
+	label << name() << "<" << _tag << ">[" << v.parent()->name() << "]";
+	return label.str();
+}
+
+
+complex<double>
+KPiSGLASS::amp(const isobarDecayVertex& v)
+{
+	// get mass
+	const double M = v.parent()->lzVec().M();                 // parent mass
+
+	complex<double> ampl = 0.0;
+
+	if (M > _kaonChargedMass+_piChargedMass and M < _MMax) {
+		const double q = breakupMomentum(M, _kaonChargedMass, _piChargedMass);
+		const double q0 = breakupMomentum(_M0, _kaonChargedMass, _piChargedMass);
+
+		const double G = _G0 * _M0 / M * q / q0;
+		const double deltaR = atan(_M0 * G / (_M0 * _M0 - M * M));
+
+		const double deltaF = atan(2.0 * _a * q / (2.0 + _a * _r * q * q));
+
+		ampl = _F * sin(deltaF + _phiF)    * exp(complex<double>(0.0, deltaF + _phiF))
+		     + _R * sin(deltaR + _phiRsin) * exp(complex<double>(0.0, deltaR + _phiR)) * exp(complex<double>(0.0, 2 * (deltaF + _phiF)));
+		ampl *= M/q; // divide by Kpi phase-space
+	}
+
+	return ampl;
+
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+KPiSPalanoPennington::KPiSPalanoPennington(const double MMax)
```

Comment:
Same comments as for KPiSGLASS::KPiSGLASS() in #L621 ff.

Reply:
Done in the header file and in the cc file (for the parameters)